### DISCUSSION TITLE
PS/2  keyboard

### DISCUSSION THREAD

**Discussion by jgalinat** [thehighnibble/thehighnibble.github.io#79] [S-132 add-on for IMSAI & Cromemco replicas]
Hi, I completed my S-132 over the weekend and it definitely is cool!  However, what had me stumped initially was keyboard input was not working. I tried 6 different USB keyboards and only one worked but unfortunately that one has several flaky keys.  From reading the discussions, I see that I am not the only one that had problems finding a compatible USB KB. 
  I do have several PS/2 keyboards  along with the proper PS/2 connector for the PCB. I am thinking of removing the USB HID PCB and just soldering in the PS/2 connector.  I read the BOM and build instructions and didn’t see any configuration changes but I wanted to check with other builders first. 
  I can’t see buying another USB KB that may or not work. 
  Thank you for any advice you can provide. 
  regards,
Jeff Galinat

**Comment by jgalinat** [comment]
Problem solved for the time being. I did another search through my stash and found a good keyboard that works with the S-132.   I would still appreciate an answer on my original post if anyone knows regarding adding a PS/2 connector just in case I ever have to go that route.

**Comment by dmcnaugh** [comment]
Jeff,
I'm glad you found a compatible keyboard in your collection.
Although this is a topic a few people have posted about I would not call it a common problem. There are hundreds of people running VT-132 and S-132 kits with this USB-HID keyboard adapter that have not had any problems. The few people that have had a problem have posted about it.

What makes a keyboard incompatible is if it has a built-in USB hub, or typically, keyboards with additional keys for things like dedicated multimedia controls or system controls (on/off/standby etc...) as this usually implies an internal hub to seperate the standard USB keyboard interface and the additional controls.

I was surprised to hear you has a series of failures, so it would be useful if you could email with details of each keyboard that didn't work so I can see if I can figure it out.

But the general rule is: the simpler (and often cheaper) the USB keyboard the better. The $12 USB wired or $24 wireless keyboard from Staples is a good measure.

In answer to your question RE: PS/2 connector the answer is YES. If you remove the USB-HID interface (red daughter PCB) and solder in the 2 x 120 ohm, 2 x 2K ohm and PS/2 keyboard connector then it will just work. The firmware detects the keyboard type at startup. 
NOTE: you are just as likely to have keyboard compatibility issues if you try to use a passive USB->PS/2 connector with this. A keyboard with an actual PS/2 plug on it should always work, but not all USB keyboards are able to interface with a PS/2 connector with a passive adapter.

**Reply by jgalinat** [reply]
Hi David,
  Thanks for the reply & info on the PS/2 interface. I had read a post by John (GrantMeStrength) than had mentioned that keyboards with hubs probably wouldn’t work.  I did have 3 with hubs so I tried them anyway and as I anticipated, they did not work. I then tried a Lenovo SK-8825 and a Dell SK-8115 which are both pretty basic keyboards (i.e no hubs or multimedia keys) and they also didn’t work.   The first one that worked was a cheap Softkey flexible rubber keyboard designed for harsh environments. The trouble with that KB was some of the keys were flaky. I then found a Havit AS-2188 compact KB that worked fine.   One thing I noticed was on the working KBs was the LEDs blinked on power up while on the non working KBs there was no LED activity. 
  
  All the PS/2 keyboards I have are true PS/2 interface with no adapters. 
  
I have posted pictures of the Lenovo & Dell KBs that didn’t work along with the Softkey and Havit KBs that did work. 

![2E276F23-7741-4EFC-9DEF-BFF4A2DC1B9D](https://github.com/thehighnibble/thehighnibble.github.io/assets/131741410/e944b286-2064-4ae9-89ef-ba74e2ffb826)

![C412A857-4544-4520-BD4D-B041691BC2DF](https://github.com/thehighnibble/thehighnibble.github.io/assets/131741410/38020749-c0e5-488e-a216-fdd3912c0565)

![46D57972-FA32-4A0B-B0CE-FCD0D4CAF0EA](https://github.com/thehighnibble/thehighnibble.github.io/assets/131741410/b881888f-fc3e-4e84-8bda-f22b66016146)

**Reply by jgalinat** [reply]
Well, I made my weekly trip to my local electronics surplus store with with adjacent new electronics store.  I had made the trip for the surplus store but I went next door and found a cheap $10 USB KB. Just like others said, the cheaper KBs work well and this one worked fine for the S-132. 
![4DC7BE0B-8013-4062-837E-2FE75BECD1C4](https://github.com/thehighnibble/thehighnibble.github.io/assets/131741410/783c9b90-f3ca-4f7f-8cec-820676c1c0e4)
![836FAC62-D99B-4F18-8759-44855F2B69E2](https://github.com/thehighnibble/thehighnibble.github.io/assets/131741410/8a6c08a0-717e-4062-af40-f0cf4cef48fb)

**Comment by jgalinat** [comment]
Thanks for the help. Closing the topic.
